src/Language/Docker/Parser/Instruction.hs

```diff
@@ -54,44 +55,77 @@ parseArg = do
       df <- untilEol "the argument value"
       return $ Arg name (Just df)
 
-parseUser :: Parser (Instruction Text)
+parseUser :: (?esc :: Char) => Parser (Instruction Text)
 parseUser = do
   reserved "USER"
   username <- untilEol "the user"
   return $ User username
 
-parseWorkdir :: Parser (Instruction Text)
+parseWorkdir :: (?esc :: Char) => Parser (Instruction Text)
 parseWorkdir = do
   reserved "WORKDIR"
   directory <- untilEol "the workdir path"
   return $ Workdir directory
 
-parseVolume :: Parser (Instruction Text)
+parseVolume :: (?esc :: Char) => Parser (Instruction Text)
 parseVolume = do
   reserved "VOLUME"
   directory <- untilEol "the volume path"
   return $ Volume directory
 
-parseMaintainer :: Parser (Instruction Text)
+parseMaintainer :: (?esc :: Char) => Parser (Instruction Text)
 parseMaintainer = do
   reserved "MAINTAINER"
   name <- untilEol "the maintainer name"
   return $ Maintainer name
 
-parseEntryPoint :: Parser (Instruction Text)
+parseEntryPoint :: (?esc :: Char) => Parser (Instruction Text)
 parseEntryPoint = do
   reserved "ENTRYPOINT"
   Entrypoint <$> arguments
 
-parseOnbuild :: Parser (Instruction Text)
+parseOnbuild :: (?esc :: Char) => Parser (Instruction Text)
 parseOnbuild = do
   reserved "ONBUILD"
   OnBuild <$> parseInstruction
 
-parseComment :: Parser (Instruction Text)
-parseComment = Comment <$> comment
+parsePragma :: (?esc :: Char) => Parser (Instruction Text)
+parsePragma = do
+  void $ char '#'
+  void onlySpaces
+  (try parseEscapePragma <?> "the escape")
+    <|> (try parseSyntaxPragma <?> "the syntax")
```

Comment:
I believe you don't need either `try`

Reply:
also, can you please use `choice` instead of `<|>`. I find it more approachable

Reply:
You are right. The `try` is superfluous since the entire `parsePragma` call is `try`ed in `parseComment` already. `choice` is good too, although I don't find `<|>` that horrible in this context.